presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -2324,10 +2324,15 @@ VeloxQueryPlanConverterBase::toVeloxQueryPlan(
     limit = *node->maxRowCountPerPartition;
   }
 
+  std::optional<std::string> rowNumberColumnName;
+  if (!node->partial) {
+    rowNumberColumnName = node->rowNumberVariable.name;
+  }
+
   return std::make_shared<core::RowNumberNode>(
       node->id,
       partitionFields,
-      node->rowNumberVariable.name,
+      rowNumberColumnName,
```

Comment:
Nit : You can inline here instead of creating a separate variable

node->partial ? std::nullopt : node->rowNumberVariable.name,

Reply:
I'll change this for TopnRowNumber node then as well. I used the same code.

Reply:
Actually, this gives a compile error
```
/Users/czentgr/gitspace/presto/presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:2330:21: error: incompatible operand types ('const std::nullopt_t' and 'const facebook::presto::protocol::String' (aka 'const basic_string<char, char_traits<char>, allocator<char>>'))
      node->partial ? std::nullopt : node->rowNumberVariable.name,
                    ^ ~~~~~~~~~~~~   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/czentgr/gitspace/presto/presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:2354:21: error: incompatible operand types ('const std::nullopt_t' and 'const facebook::presto::protocol::String' (aka 'const basic_string<char, char_traits<char>, allocator<char>>'))
      node->partial ? std::nullopt : node->rowNumberVariable.name,
                    ^ ~~~~~~~~~~~~   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
```
I think it doesn't work because the argument is a reference.
```
const std::optional<std::string>& rowNumberColumnName,
```
So it needs a local variable to reference.

Reply:
Either way, removing the reference will incur a copy. But that already occurs when assigning to the local variable (if a string was provided). Thus, removing the variable and changing the parameters to not use a reference still incurs a copy but saves the variable.

Reply:
And this would require another velox change. This is in the Velox code. So perhaps I'll keep the original and it can be changed later?

Reply:
That's alright. We can stay with the variable for now.

Reply:
> And this would require another velox change. This is in the Velox code. So perhaps I'll keep the original and it can be changed later?

Lets stick with the current code. There isn't any need for Velox follow-up since its inconsistent in PlanNode classes.